static/app/components/modals/newDemoSignUp.tsx

```diff
@@ -0,0 +1,126 @@
+import {css} from '@emotion/react';
+import styled from '@emotion/styled';
+
+import habitsSuccessfulCustomer from 'sentry-images/spot/habitsSuccessfulCustomer.jpg';
+
+import {ModalRenderProps} from 'sentry/actionCreators/modal';
+import Button from 'sentry/components/button';
+import ButtonBar from 'sentry/components/buttonBar';
+import HighlightCornerContainer from 'sentry/components/highlightCornerModal';
+import {t} from 'sentry/locale';
+import space from 'sentry/styles/space';
+import trackAdvancedAnalyticsEvent from 'sentry/utils/analytics/trackAdvancedAnalyticsEvent';
+import {
+  extraQueryParameter,
+  extraQueryParameterWithEmail,
+  urlAttachQueryParams,
+} from 'sentry/utils/demoMode';
+
+type Props = ModalRenderProps;
+
+const NewDemoSignUpModal = ({closeModal}: Props) => {
+  const signupUrl = urlAttachQueryParams(
+    'https://sentry.io/signup/',
+    extraQueryParameterWithEmail()
```

Comment:
@roggenkemper we should also add some referrer value here like `referrer=sandbox-walkthrough`